x-pack/plugins/monitoring/public/components/metricbeat_migration/flyout/flyout.js

```diff
@@ -0,0 +1,345 @@
+    // We attempt to provide a better UX for the user by automatically rechecking
+    // the status of their current step, once they have initiated a check manually.
+    // The logic here aims to remove the recheck one they have moved on from the
+    // step
+
+    const thisActiveStep = this.state.activeStep;
+    const nextActiveStep = nextState.activeStep;
+    const nextEnableMbStatus = nextState.checkedStatusByStep[INSTRUCTION_STEP_ENABLE_METRICBEAT];
+    const nowEnableMbStatus = this.state.checkedStatusByStep[INSTRUCTION_STEP_ENABLE_METRICBEAT];
+    const nextDisableInternalStatus = nextState.checkedStatusByStep[INSTRUCTION_STEP_DISABLE_INTERNAL];
+    const nowDisableInternalStatus = this.state.checkedStatusByStep[INSTRUCTION_STEP_DISABLE_INTERNAL];
+
+    const setupInterval = (nextEnableMbStatus && !nowEnableMbStatus) || (nextDisableInternalStatus && !nowDisableInternalStatus);
+    const removeInterval = thisActiveStep !== nextActiveStep;
+    if (removeInterval) {
+      clearInterval(this.checkInterval);
+      this.clearInterval = null;
+    }
+
+    if (setupInterval) {
+      this.checkInterval = setInterval(async () => {
+        await this.checkForMigrationStatus();
+      }, AUTO_CHECK_INTERVAL_IN_MS);
+    }
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.checkInterval);
+  }
+
+  checkForMigrationStatus = async () => {
+    this.setState({ checkingMigrationStatus: true });
+    await ensureMinimumTime(this.props.updateProduct(), 1000);
+    this.setState(state => ({
+      ...state,
+      checkingMigrationStatus: false,
+      checkedStatusByStep: {
+        ...state.checkedStatusByStep,
+        [this.state.activeStep]: true,
+      }
+    }));
+  }
+
+  setEsMonitoringUrl = esMonitoringUrl => {
+    storage.set(ES_MONITORING_URL_KEY, esMonitoringUrl);
+    this.setState({ esMonitoringUrl });
+  }
+
+  renderActiveStep() {
+    const { product, productName, onClose, meta, instance } = this.props;
+    const {
+      activeStep,
+      esMonitoringUrl,
+      checkedStatusByStep,
+      checkingMigrationStatus,
+    } = this.state;
+
+    switch (activeStep) {
+      case INSTRUCTION_STEP_SET_MONITORING_URL:
+        return (
+          <EuiForm>
+            <EuiFormRow
+              fullWidth
+              label={i18n.translate('xpack.monitoring.metricbeatMigration.flyout.step1.monitoringUrlLabel', {
+                defaultMessage: 'Monitoring cluster URL'
+              })}
+              helpText={i18n.translate('xpack.monitoring.metricbeatMigration.flyout.step1.monitoringUrlHelpText', {
+                defaultMessage: `This is typically a single instance, but if you have multiple, enter all of instance urls comma-separated.
+                Keep in mind that the running metricbeat instance will need to be able to communicate with these Elasticsearch servers.`
+              })}
+            >
+              <EuiFieldText
+                value={esMonitoringUrl}
```

Comment:
You'll need `fullWidth` on the input as well

Reply:
<img width="964" alt="Screen Shot 2019-06-14 at 9 59 14 AM" src="https://user-images.githubusercontent.com/56682/59514582-2989b600-8e8b-11e9-80d0-2fda9f0cd3a5.png">